ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java

```diff
@@ -59,6 +66,133 @@ public void setUp() {
+                .version("1")
+                .protocol("http")
+                .actions(Arrays.asList(predictAction))
+                .build();
+            HttpJsonConnectorExecutor executor = spy(new HttpJsonConnectorExecutor(connector));
+            when(executor.getHttpClient()).thenReturn(httpClient);
+            MLInputDataset inputDataSet = TextDocsInputDataSet.builder().docs(Arrays.asList("test doc1", "test doc2")).build();
+            ModelTensorOutput modelTensorOutput = executor.executePredict(MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build());
+            verify(executor, Mockito.times(1)).invokeRemoteModel(any(), any(), any(), any());
+            Assert.assertEquals(1, modelTensorOutput.getMlModelOutputs().size());
+            Assert.assertEquals("response", modelTensorOutput.getMlModelOutputs().get(0).getMlModelTensors().get(0).getName());
+        } catch (Exception e) {
+            fail("An exception was thrown: " + e.getMessage());
+        }
+    }
+
+    public void invokeRemoteModel_HeaderTest(Map<String, String> header) throws ClientProtocolException, IOException {
+        ConnectorAction predictAction = ConnectorAction.builder()
+            .actionType(ConnectorAction.ActionType.PREDICT)
+            .method("GET")
+            .headers(header)
+            .url("http://test.com/mock")
+            .requestBody("{\"input\": ${parameters.input}}")
+            .build();
+        Map<String, String> credential = new HashMap<>();
+        credential.put("key", "test_key_value");
+        when(httpClient.execute(any())).thenReturn(response);
+        HttpEntity entity = new StringEntity("{\"response\": \"test result\"}");
+        when(response.getEntity()).thenReturn(entity);
+         HttpConnector connector = spy(HttpConnector.builder()
+                .name("test_connector_name")
+                .description("this is a test connector")
+                .version("1")
+                .protocol("http")
+                .credential(credential)
+                .actions(Arrays.asList(predictAction))
+                .backendRoles(Arrays.asList("role1", "role2"))
+                .accessMode(AccessMode.PUBLIC)
+                .build());
+        Function<String, String> decryptFunction = (n) -> (n);
+        connector.decrypt(decryptFunction);
+        HttpJsonConnectorExecutor executor = spy(new HttpJsonConnectorExecutor(connector));
+        when(executor.getHttpClient()).thenReturn(httpClient);
+        MLInputDataset inputDataSet = TextDocsInputDataSet.builder().docs(Arrays.asList("test doc1", "test doc2")).build();
+        ModelTensorOutput modelTensorOutput = executor.executePredict(
+            MLInput.builder().algorithm(FunctionName.REMOTE)
+            .inputDataset(inputDataSet)
+            .build()
+        );
+        verify(connector, Mockito.times(1)).getDecryptedHeaders();
+        Assert.assertEquals(header, executor.getConnector().getDecryptedHeaders());
```

Comment:
This way we are just checking two maps are the same object in memory. We should check the content of the map.